core/java/iesi-rest-without-microservices/src/main/java/io/metadew/iesi/server/rest/executionrequest/ExecutionRequestDtoRepository.java

```diff
@@ -22,317 +23,159 @@
-                    "on script_execution_requests.SCRPT_REQUEST_ID = file_script_execution_requests.SCRPT_REQUEST_ID " +
-                    "left outer join " + MetadataTablesConfiguration.getInstance().getMetadataTableNameByLabel("ScriptNameExecutionRequests").getName() + " name_script_execution_requests " +
-                    "on script_execution_requests.SCRPT_REQUEST_ID = name_script_execution_requests.SCRPT_REQUEST_ID " +
-                    "inner join " + MetadataTablesConfiguration.getInstance().getMetadataTableNameByLabel("ScriptExecutions").getName() + " script_executions " +
-                    "on script_executions.SCRPT_REQUEST_ID = script_execution_requests.SCRPT_REQUEST_ID " +
-                    "where execution_requests.REQUEST_ID = {0};";
+    private String getBaseQuery(Pageable pageable, List<ExecutionRequestFilter> executionRequestFilters) {
+        return "SELECT distinct execution_requests.REQUEST_ID, execution_requests.REQUEST_TMS, name_script_execution_requests.SCRPT_NAME, name_script_execution_requests.SCRPT_VRS " +
+                "FROM " + MetadataTablesConfiguration.getInstance().getMetadataTableNameByLabel("ExecutionRequests").getName() + " execution_requests " +
+                "left outer join " + MetadataTablesConfiguration.getInstance().getMetadataTableNameByLabel("ScriptExecutionRequests").getName() + " script_execution_requests " +
+                "on execution_requests.REQUEST_ID = script_execution_requests.ID " +
+                "left outer join " + MetadataTablesConfiguration.getInstance().getMetadataTableNameByLabel("ScriptFileExecutionRequests").getName() + " file_script_execution_requests " +
+                "on script_execution_requests.SCRPT_REQUEST_ID = file_script_execution_requests.SCRPT_REQUEST_ID " +
+                "left outer join " + MetadataTablesConfiguration.getInstance().getMetadataTableNameByLabel("ScriptNameExecutionRequests").getName() + " name_script_execution_requests " +
+                "on script_execution_requests.SCRPT_REQUEST_ID = name_script_execution_requests.SCRPT_REQUEST_ID " +
+                "left outer join " + MetadataTablesConfiguration.getInstance().getMetadataTableNameByLabel("ExecutionRequestLabels").getName() + " execution_request_labels " +
+                "on execution_requests.REQUEST_ID = execution_request_labels.REQUEST_ID " +
+                getWhereClause(executionRequestFilters) +
+                getOrderByClause(pageable) +
+                getLimitAndOffsetClause(pageable);
+    }
 
-    private final MetadataRepositoryConfiguration metadataRepositoryConfiguration;
 
+    private String getWhereClause(List<ExecutionRequestFilter> executionRequestFilters) {
+        String filterStatements = executionRequestFilters.stream().map(executionRequestFilter -> {
```

Comment:
You don't need to filter by date?

Reply:
not at the moment. Maybe future filtering can be done on request timestamp. Filtering would then occur on a time range (from - to). I created a dedicated issue for this #200